src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -957,14 +961,20 @@ internal static void AddSubjectClaims(
             nbfSet |= nbfReset;
         }
 
+        internal static bool IncludeDefaultHeaderClaim(string claim, ISet<string> excludedDefaultHeaderClaims) =>
+            excludedDefaultHeaderClaims is null ||
+            excludedDefaultHeaderClaims.Count == 0 ||
+            !excludedDefaultHeaderClaims.Contains(claim);
+
         internal static void WriteJwsHeader(
```

Comment:
The number of parameters is getting excessive.

Suggestion: always use the SecurityTokenDescriptor, we would allocate on the code path that doesn't pass in an object, but if the user wants to avoid the allocation, they can use the overload that does accept a SecurityTokenDescriptor

Reply:
I think this should be a separate PR, but I agree.

Reply:
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2991